metaflow/plugins/kfp/kfp.py

```diff
@@ -979,6 +983,43 @@ def build_kfp_dag(
                     "get_workflow_uid"
                 )
 
+            s3_sensor_op = None
+            s3_sensor_deco = self.flow._flow_decorators.get("s3_sensor")
+            if s3_sensor_deco:
+                path = s3_sensor_deco.path
+                timeout_seconds = s3_sensor_deco.timeout_seconds
+                polling_interval_seconds = s3_sensor_deco.polling_interval_seconds
+                func = s3_sensor_deco.func
+
+                # see https://github.com/kubeflow/pipelines/pull/1946/files
+                # KFP does not support the serialization of Python functions directly. The KFP team took
+                # the approach of using base64 encoding + pickle. Pickle didn't quite work out
+                # in this case because pickling a function directly stores references to the function's path,
+                # which couldn't be resolved when the formatter function was unpickled within the running
+                # container. Instead, we took the approach of marshalling just the code of the formatter
+                # function, and reconstructing the function within the kf_s3_sensor.py code.
+                func_code_encoded = base64.b64encode(
+                    marshal.dumps(func.__code__)
+                ).decode("ascii")
+
+                s3_sensor_op = func_to_container_op(
+                    wait_for_s3_path,
+                    base_image="hsezhiyan/s3_sensor:1.0",
+                )(
+                    path=path,
+                    timeout_seconds=timeout_seconds,
+                    polling_interval_seconds=polling_interval_seconds,
+                    func_code_encoded=func_code_encoded,
+                    flow_parameters_json=flow_parameters_json,
+                ).set_display_name(
+                    "s3_sensor"
+                )
+                # tighten the resources so customers don't bear large costs
+                s3_sensor_op.container.set_cpu_request("0.1")
```

Comment:
@talebzeghmi want to bring your attention to this piece of code. It was required by Rajesh that we stringently lock down the resources so customers don't bear large cost waiting for an S3 path. Does this seem reasonable?

Also, I noticed you didn't do anything like it for the `workflow_uid_op`.  I'm guessing it's because `workflow_uid_op` is relatively short-lived?

Reply:
These resource requirements makes sense to me. Try making these lines a (maybe private?) function so that it is reusable for `workflow_uid_op` to use the same resources.